crates/uv-python/src/downloads.rs

```diff
@@ -446,11 +472,113 @@ impl ManagedPythonDownload {
 
     /// Iterate over all [`ManagedPythonDownload`]s.
     pub fn iter_all() -> impl Iterator<Item = &'static ManagedPythonDownload> {
-        PYTHON_DOWNLOADS
-            .iter()
-            // TODO(konsti): musl python-build-standalone builds are currently broken (statically
-            // linked), so we pretend they don't exist. https://github.com/astral-sh/uv/issues/4242
-            .filter(|download| download.key.libc != Libc::Some(target_lexicon::Environment::Musl))
+        PYTHON_DOWNLOADS.get_or_init(|| {
+            if let Ok(json_path) = std::env::var("UV_PYTHON_DOWNLOADS_JSON_PATH") {
+                let file = match fs_err::File::open(&json_path) {
+                    Ok(file) => file,
+                    Err(e) => {
+                        panic!("Failed to open Python downloads JSON file at {json_path}: {e}");
+                    }
+                };
+
+                let json_downloads: HashMap<String, JsonPythonDownload> = serde_json::from_reader(file)
+                    .unwrap_or_else(|e| {
+                        panic!("Failed to parse JSON file at {json_path}: {e}");
+                    });
```

Comment:
Maybe a bit of a silly question, but is there a reason we shouldn't deserialize directly into `ManagedPythonDownload`? Why the intermediate type?

Reply:
TBH, that's what the AI suggested, but I had the same thought as you, so I tried to derive the `Deserialize` on `ManagedPythonDownload` and then on `PythonInstallationKey`. I understood that `target_lexicon's enums do not implement the `Deserialize` trait, and even if I overcome this, the conversion from JSON will not be as is. So the only way to improve it (the way I see it) is to implement `Deserialize` on our own and define the JSON structs inside the implementation.

If you see a better way, please let me know

Reply:
It looks like there's a `serde` feature for `target_lexicon`, did you try that?

Reply:
What part is not "as-is"? Like some custom mapping of architectures or something?

Reply:
> It looks like there's a serde feature for target_lexicon, did you try that?

from the [code](https://github.com/bytecodealliance/target-lexicon/blob/c3f72c4ced783523aba166d1c52f1b355de8c81a/src/lib.rs#L67), I understood that they only implemented it for the `Triple` struct, so it not useful here :(

> What part is not "as-is"? Like some custom mapping of architectures or something?

yes, it's small details, like the `implementation` that will need special implementation, or the `arch`. the main problem here is the structs of `target_lexicon`